internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -0,0 +1,679 @@
+	}
+	updatedPhase := updateNaBSLPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseCreated)
+
+	if bslCondition || updatedPhase {
+		if updateErr := r.Status().Update(ctx, nabsl); updateErr != nil {
+			logger.Error(updateErr, failedUpdateStatusError)
+			return false, updateErr
+		}
+	}
+
+	return false, nil
+}
+
+// syncStatus
+func (r *NonAdminBackupStorageLocationReconciler) syncStatus(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	veleroObjectsNACUUID := nabsl.Status.VeleroBackupStorageLocation.NACUUID
+
+	// Check if VeleroBackupStorageLocation already exists
+	veleroBsl, err := function.GetVeleroBackupStorageLocationByLabel(ctx, r.Client, r.OADPNamespace, veleroObjectsNACUUID)
+	if err != nil {
+		logger.Error(err, "Failed to get VeleroBackupStorageLocation", constant.UUIDString, veleroObjectsNACUUID)
+		return false, err
+	}
+
+	// Ensure that the NonAdminBackup's NonAdminBackupStatus is in sync
+	// with the VeleroBackup. Any required updates to the NonAdminBackup
+	// Status will be applied based on the current state of the VeleroBackup.
+	updated := updateNaBSLVeleroBackupStorageLocationStatus(&nabsl.Status, veleroBsl)
+	if updated {
+		if err := r.Status().Update(ctx, nabsl); err != nil {
+			logger.Error(err, "Failed to update NonAdminBackupStorageLocation Status after VeleroBackupStorageLocation reconciliation")
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackupStorageLocation Status updated successfully")
+	} else {
+		logger.V(1).Info("NonAdminBackup Status unchanged")
+	}
+
+	return false, nil
+}
+
+// updateNaBSLVeleroBackupStorageLocationStatus sets the VeleroBackupStorageLocation status field in NonAdminBackupStorageLocation object status and returns true
+// if the VeleroBackupStorageLocation fields are changed by this call.
+func updateNaBSLVeleroBackupStorageLocationStatus(status *nacv1alpha1.NonAdminBackupStorageLocationStatus, veleroBackupStorageLocation *velerov1.BackupStorageLocation) bool {
+	if status == nil || veleroBackupStorageLocation == nil {
+		return false
+	}
+	if status.VeleroBackupStorageLocation == nil {
+		status.VeleroBackupStorageLocation = &nacv1alpha1.VeleroBackupStorageLocation{}
+	}
+
+	// Treat nil as equivalent to a zero-value struct
+	currentStatus := velerov1.BackupStorageLocationStatus{}
+	if status.VeleroBackupStorageLocation.Status != nil {
+		currentStatus = *status.VeleroBackupStorageLocation.Status
+	}
+
+	// Return false if both statuses are equivalent
+	if reflect.DeepEqual(currentStatus, veleroBackupStorageLocation.Status) {
+		return false
+	}
+
+	// Update and return true if they differ
+	status.VeleroBackupStorageLocation.Status = veleroBackupStorageLocation.Status.DeepCopy()
+	return true
+}
+
+// updateNaBSLPhase updates the phase of the NonAdminBackupStorageLocation
+func updateNaBSLPhase(phase *nacv1alpha1.NonAdminBackupStorageLocationPhase, newPhase nacv1alpha1.NonAdminBackupStorageLocationPhase) bool {
```

Comment:
what is the difference between this and `updateNonAdminBslPhase`?

Reply:
I will use one from nonadminbackup_controller.go. Ideally that function syould be part of function.go, but that's not something I will do in this PR as this will modify other controllers which I don't want to perform in this PR